src/Generators/NovaTestGenerator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace RonasIT\Support\Generators;
+
+use Illuminate\Support\Str;
+use Laravel\Nova\NovaServiceProvider;
+use RonasIT\Support\Events\SuccessCreateMessage;
+use RonasIT\Support\Exceptions\ClassAlreadyExistsException;
+use RonasIT\Support\Exceptions\ClassNotExistsException;
+
+class NovaTestGenerator extends AbstractTestsGenerator
+{
+    public function generate(): void
+    {
+        if (class_exists(NovaServiceProvider::class)) {
+            if (!$this->classExists('nova', "{$this->model}")) {
+                $this->throwFailureException(
+                    ClassNotExistsException::class,
+                    "Cannot create Nova{$this->model}Test cause {$this->model} Nova resource does not exist.",
+                    "Create {$this->model} Nova resource."
+                );
+            }
+
+            if ($this->classExists('nova', "Nova{$this->model}Test")) {
+                $this->throwFailureException(
+                    ClassAlreadyExistsException::class,
+                    "Cannot create Nova{$this->model}Test cause it's already exist.",
+                    "Remove Nova{$this->model}Test."
+                );
+            }
+
+            parent::generate();
+        } else {
+            event(new SuccessCreateMessage("Nova is not installed and NovaTest is skipped"));
+        }
+    }
+
+    public function generateTests(): void
+    {
+        $actions = [];
+
+        if (file_exists(base_path($this->paths['nova_actions']))) {
+            $objectsInsideFolder = scandir(base_path($this->paths['nova_actions']));
+            $modelActions = array_filter($objectsInsideFolder, function ($value) {
```

Comment:
it is not the best way to determine the action relation. I believe it would be better to get actions classes from the NovaResource